storage/buckets/buckets_test.go

```diff
@@ -326,6 +370,78 @@ func TestLifecycleManagement(t *testing.T) {
 	}
 }
 
+func TestBucketLabel(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	ctx := context.Background()
+	testutil.CleanBucket(ctx, t, tc.ProjectID, bucketName)
+
+	testutil.Retry(t, 10, 10*time.Second, func(r *testutil.R) {
+		if err := addBucketLabel(ioutil.Discard, bucketName, "label-name", "label-value"); err != nil {
+			r.Errorf("addBucketLabel: %v", err)
+		}
+	})
+	if err := removeBucketLabel(ioutil.Discard, bucketName, "label-name"); err != nil {
+		t.Fatalf("removeBucketLabel: %v", err)
+	}
+}
+
+func TestDefineBucketWebsiteConfiguration(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	ctx := context.Background()
+	testutil.CleanBucket(ctx, t, tc.ProjectID, bucketName)
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	index := "index.html"
+	notFoundPage := "404.html"
+	if err := defineBucketWebsiteConfiguration(ioutil.Discard, bucketName, index, notFoundPage); err != nil {
+		t.Fatalf("defineBucketWebsiteConfiguration: %v", err)
+	}
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		t.Fatalf("Bucket(%q).Attrs: %v", bucketName, err)
+	}
+	if attrs.Website.MainPageSuffix != index {
+		t.Fatalf("got index page: %v, want %v", attrs.Website.MainPageSuffix, index)
+	}
+	if attrs.Website.NotFoundPage != notFoundPage {
+		t.Fatalf("got not found page: %v, want %v", attrs.Website.NotFoundPage, notFoundPage)
+	}
+}
+
+func TestSetBucketPublicIAM(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	ctx := context.Background()
+	testutil.CleanBucket(ctx, t, tc.ProjectID, bucketName)
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	policy, err := client.Bucket(bucketName).IAM().Policy(ctx)
+	if err != nil {
+		t.Fatalf("Bucket(%q).IAM().Policy: %v", bucketName, err)
+	}
+	if err := setBucketPublicIAM(ioutil.Discard, bucketName, *policy); err != nil {
+		t.Fatalf("setBucketPublicIAM: %v", err)
+	}
+	if !policy.HasRole(iam.AllUsers, iam.RoleName("roles/storage.objectViewer")) {
```

Comment:
Re-fetch policy to check here, rather than using the one you sent.

Reply:
Done